### PR TITLE
Manual supercells

### PR DESCRIPTION
**Pull request by QuantumChemist**
Manual supercells can now be set by a supercell_matrix passed via the supercell_settings dict

- [x] add unit test
- [x] add documentation
- [x] check ALGO = Normal
- [x] switch from deprecated MLIP PhononMaker setup to the new one (also check elastic tensor)

### CODE REVIEW

**Comment by QuantumChemist** [comment]
@JaGeo if you like this idea, I will add a unit test and add info to the documentation

**Comment by QuantumChemist** [comment]
but maybe I sould make it that two supercell_matrices ca be passed

**Comment by JaGeo** [comment]
I think you should add a list of supercells for each structure and benchmark structure and check that the lists are having the same size

**Review by JaGeo — APPROVED** [review]

File: autoplex/auto/phonons/jobs.py [hunk 1]
```diff
@@ -327,7 +327,9 @@ def dft_phonopy_gen_data(
     jobs = []
     dft_phonons_output = {}
     dft_phonons_dir_output = []
-    supercell_matrix = reduce_supercell_size(structure, **supercell_settings)
+    supercell_matrix = supercell_settings.get("supercell_matrix")
```
Comment (JaGeo) [inline autoplex/auto/phonons/jobs.py:330]: either add the supercell_matrix to each of the jobs and then figure out which structure in the list the supercell matrix belongs to or just count the structures. Same for the random_structure_generation

Comment (QuantumChemist) [reply autoplex/auto/phonons/jobs.py:330]: sounds good. I guess I was thinking this in the wrong direction.

File: autoplex/auto/phonons/jobs.py [hunk 2]
```diff
@@ -327,7 +327,9 @@ def dft_phonopy_gen_data(
     jobs = []
     dft_phonons_output = {}
     dft_phonons_dir_output = []
-    supercell_matrix = reduce_supercell_size(structure, **supercell_settings)
+    supercell_matrix = supercell_settings.get(structure, {}).get("supercell_matrix")
```
Comment (QuantumChemist) [inline autoplex/auto/phonons/jobs.py:330]: @JaGeo , I could make it that for each structure one could pass a supercell_matrix like:
```python
supercell_settings = {
    structure: {
        "supercell_matrix": [[0, 2, 0], [0, 0, 2], [2, 0, 0]]
    },
    "min_length": 12,  
    "max_length": 25  
}
```

Comment (QuantumChemist) [reply autoplex/auto/phonons/jobs.py:330]: Oh, the Structure object is unhashable of course...

Comment (JaGeo) [reply autoplex/auto/phonons/jobs.py:330]: Just use a list or use the mpid. And then use the number in the list/mpid to identify the supercell

Comment (QuantumChemist) [reply autoplex/auto/phonons/jobs.py:330]: I switched to the mpids:
```python
supercell_settings = {
    mp_id: {
        "supercell_matrix": [[0, 2, 0], [0, 0, 2], [2, 0, 0]]
    },
    "min_length": 12,  
    "max_length": 25  
}
```
@JaGeo  what do you think about that?

Comment (QuantumChemist) [reply autoplex/auto/phonons/jobs.py]: Would you like me to add or change something? I think this is the best option to make the user set each supercell_matrix individually if they want to have a different one.

Comment (JaGeo) [reply autoplex/auto/phonons/jobs.py]: Yes, agreed.

Comment (QuantumChemist) [reply autoplex/auto/phonons/jobs.py]: Ok great, then I will add a unit test and documentation 😃

**Comment by QuantumChemist** [comment]
Hey @JaGeo , I would say this PR is ready to be merged 😃

File: autoplex/data/phonons/flows.py [hunk 3]
```diff
@@ -625,13 +630,54 @@ def make_from_ml_model(
...
+                "bulk_relax_maker": ForceFieldRelaxMaker(
+                    relax_cell=True,
+                    relax_kwargs={"interval": 500},
+                    force_field_name="M3GNet",
+                ),
+                "phonon_displacement_maker": ForceFieldStaticMaker(
+                    name="m3gnet phonon static", force_field_name="M3GNet"
+                ),
+                "static_energy_maker": ForceFieldStaticMaker(force_field_name="M3GNet"),
+                "default_calculator_kwargs": {},
```
Comment (JaGeo) [inline autoplex/data/phonons/flows.py:656]: kkurze frage hier? solltestt du da nicht die default_calculator_kwargs setzen?

Comment (QuantumChemist) [reply autoplex/data/phonons/flows.py:656]: moment

File: autoplex/data/phonons/flows.py [hunk 4]
```diff
@@ -625,13 +630,54 @@ def make_from_ml_model(
...
+                    relax_kwargs={"interval": 500},
+                    force_field_name="MACE",
+                ),
+                "phonon_displacement_maker": ForceFieldStaticMaker(
+                    name="mace phonon static", force_field_name="MACE"
+                ),
+                "static_energy_maker": ForceFieldStaticMaker(force_field_name="MACE"),
+                "default_calculator_kwargs": {
+                    "device": "cuda",
+                    "model": str(potential_file),
```
Comment (JaGeo) [inline autoplex/data/phonons/flows.py:670]: here, you have a potential_file but not for the relaxation?

**Review by JaGeo — APPROVED** [review]

**Pull request merged**